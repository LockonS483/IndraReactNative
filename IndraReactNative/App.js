/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import {
  StyleSheet, Text, View
} from 'react-native';

const App: () => React$Node = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>-Hello Indra-</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    fontSize: 25,
    fontWeight: 'bold',
  },
});

export default App;
