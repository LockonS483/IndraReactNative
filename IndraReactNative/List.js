import React, { Component } from 'react'
import { Text, View, TouchableOpacity, StyleSheet, Dimensions, ScrollView } from 'react-native'

var width = Dimensions.get('window').width;

class List extends Component {
    state = {
       names: [
          {
             id: 0,
             name: 'Basic',
          },
          {
             id: 1,
             name: 'Bacteria',
          },
          {
             id: 2,
             name: 'El Farol Bar',
          },
          {
             id: 3,
             name: 'Adam Smith\'s Fashion Model',
          },
          {
            id: 4,
            name: 'Predator-Prey Dynamics',
          },
          {
            id: 5,
            name: 'Schelling\'s Segregation Model',
          },
          {
            id: 6,
            name: 'Menger\'s Origin of Money',
          },
          {
            id: 7,
            name: 'Epidemic Model',
          },
          {
            id: 8,
            name: 'An Abelian Sandpile',
          },
          {
            id: 9,
            name: 'Wolfram\'s Rules',
          },
       ]
    }
    alertItemName = (item) => {
       alert(item.name)
    }
    render() {
       return (
          <View>
              <ScrollView>
             {
                this.state.names.map((item, index) => (
                   <TouchableOpacity
                      key = {item.id}
                      style = {styles.container}
                      onPress = {() => this.alertItemName(item)}>
                      <Text style = {styles.text}>
                         {item.name}
                      </Text>
                   </TouchableOpacity>
                ))
             }
             </ScrollView>
          </View>
       )
    }
 }
 export default List
 
 const styles = StyleSheet.create ({
    container: {
        padding: 30,
        width: width* 0.9,
        marginTop: 10,
        backgroundColor: 'white',
        alignItems: 'center',
        borderRadius: 5,
    },
    text: {
       color: '#4f603c'
    }
 })