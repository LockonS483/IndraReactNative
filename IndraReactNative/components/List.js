import React, { Component, useState } from 'react'
import { Text, View, TouchableOpacity, StyleSheet, Dimensions, ScrollView } from 'react-native'

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

class List extends Component {
    constructor(props) {
        super(props);
        this.state = {names: [], dataNames: []};
    }
    
    componentDidMount(){
        fetch('http://indrasnet.pythonanywhere.com/models', {
          method: 'GET'
        })
        .then((response) => response.json())
        .then((responseJson) => {
          this.setState({
            dataNames: responseJson
          })
          var activeNames = [];
          this.state.dataNames.forEach((item, index) => {
              if(item.active){
                  activeNames.push(item);
              }
          });
          this.setState({names : activeNames});
        })
        
    }

    alertItemName = (item) => {
       alert(item.name);
    }
    render() {
        return (
        <View>
              <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={{paddingBottom: 50}} style = {styles.scroll}>
                <Text style={styles.titleText}>-Indra Models-</Text>
                {
                    this.state.names.map((item, index) => (
                           <TouchableOpacity
                              key = {item.name}
                              style = {styles.model}
                              onPress = {() => this.alertItemName(item)}>
                                <Text style = {styles.text}>
                                    {item.name}
                                </Text>
                           </TouchableOpacity>
                    ))
                }
            </ScrollView>
        </View>
        )
    }
 }
 export default List
 
 const styles = StyleSheet.create ({
    model: {
        padding: 30,
        width: width * 0.9,
        marginTop: 8,
        backgroundColor: 'white',
        alignItems: 'center',
        borderRadius: 5,
    },
    text: {
       color: '#4f603c'
    },
    scroll: {
        marginTop: height * 0.05,
        height: '100%',
    },
    titleText: {
        fontSize: 25,
        fontWeight: 'bold',
        textAlign: 'center',
        marginBottom: 10,
      },
 })