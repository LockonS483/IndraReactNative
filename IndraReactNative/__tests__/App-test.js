/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import List from '../components/List';
import renderer from 'react-test-renderer';

jest.useFakeTimers()
it('renders correctly', () => {
  renderer.create(<App />);
});

it('Renders title and list across the screen', () => {
  const modelList = renderer.create(<List />).toJSON();
  expect(modelList).toMatchSnapshot();
});
