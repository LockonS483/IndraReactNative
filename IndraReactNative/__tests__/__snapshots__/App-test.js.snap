// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders title and list across the screen 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "height": "100%",
        "marginTop": 66.7,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        -Indra Models-
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
